packages/plugin-rust-cargo/index.js

```diff
@@ -11,8 +11,10 @@ function linkDependency(blob, key, value) {
 function processTOML(blob, plugin) {
   let results = [];
   const json = TOML.parse(blob.toString());
+  const entries = Object.assign(json['dependencies'], json['dev-dependencies']);
 
-  Object.entries(json.dependencies).forEach(([name, version]) => {
+  Object.entries(entries).forEach(([name, value]) => {
+    const version = typeof value === "string" ? value : value.version;
```

Comment:
The regex being used (`tomlRegExKeyValue`) doesn't seem to work with object values like this. I had to add another regex that handled object values like this:

```js
export function tomlRegExObjectKeyValue(key, value) {
  const regexKey = escapeRegexString(key);
  const regexValue = escapeRegexString(value);

  return new RegExp(`(${regexKey})\\s*=\\s*{.*=\\s*"(${regexValue})"`, 'g');
}
```

And then call this based on if `value` is a string or object.

This is the full change:

```diff
@ -1,21 +1,27 @@
 import * as TOML from '@iarna/toml';
 import insertLink from '@octolinker/helper-insert-link';
-import { tomlRegExKeyValue } from '@octolinker/helper-regex-builder';
+import {
+  tomlRegExKeyValue,
+  tomlRegExObjectKeyValue,
+} from '@octolinker/helper-regex-builder';
import liveResolverQuery from '@octolinker/resolver-live-query';

-function linkDependency(blob, key, value) {
-  const regex = tomlRegExKeyValue(key, value);
+function linkDependency(blob, key, value, isStringMatch) {
+  const regex = isStringMatch
+    ? tomlRegExKeyValue(key, value)
+    : tomlRegExObjectKeyValue(key, value);
  return insertLink(blob, regex, this);
}

function processTOML(blob, plugin) {
  let results = [];
  const json = TOML.parse(blob.toString());
  const entries = Object.assign(json['dependencies'], json['dev-dependencies']);

  Object.entries(entries).forEach(([name, value]) => {
-    const version = typeof value === "string" ? value : value.version;
-    const linked = linkDependency.call(plugin, blob, name, version);
+    const isString = typeof value === 'string';
+    const version = isString ? value : value.version;
+    const linked = linkDependency.call(plugin, blob, name, version, isString);

    results = results.concat(linked);
  });
@ @@
```

@stefanbuck thoughts on this?

Reply:
How about passing the `isStringMatch` to `tomlRegExKeyValue` instead of creating a new regex function.

```js
export function tomlRegExKeyValue(key, value, isString) {
  const regexKey = escapeRegexString(key);
  const regexValue = escapeRegexString(value);
  const reStr = isString ? `(${regexKey})\\s*=\\s*"(${regexValue})"`
    : `(${regexKey})\\s*=\\s*{.*=\\s*"(${regexValue})"`;

  return new RegExp(reStr, 'g');
}
```

```js
function linkDependency(blob, key, value, isString) {
  const regex = tomlRegExKeyValue(key, value, isString);
  return insertLink(blob, regex, this);
}

function processTOML(blob, plugin) {
  let results = [];
  const json = TOML.parse(blob.toString());
  const entries = Object.assign(json['dependencies'], json['dev-dependencies']);

  Object.entries(entries).forEach(([name, value]) => {
    const isString = typeof value === "string";
    const version = isString ? value : value.version;
    const linked = linkDependency.call(plugin, blob, name, version, isString);

    results = results.concat(linked);
  });

  return results;
}
```

Reply:
Sorry for the late response. I think both are valid approaches and at the end it doesn't really matter. I don't see this toml regex being used in other places than `plugin-rust-cargo/index.js`. Unless @xt0rted disagrees I think we can go ahead with the single function suggestions from @chawyehsu

Reply:
I prefer the single function, thoughts? @xt0rted

Reply:
I prefer two functions since it's clearer what's goin on, but either way is fine with me.